social/twitter/README.md

```diff
@@ -59,3 +59,9 @@ To send a Direct Message (DM) - use a payload like.
 If `msg.media` exists and is a Buffer object, this node will treat it as an image and attach it to the tweet.
 
 If `msg.params` exists and is an object of name:value pairs, this node will treat it as parameters for the update request.
+
+#### Retweet
+
+To retweet an existing tweet, set `msg.retweet` to the tweet id. The tweet id can be found in the `id_str` field of the tweet object.
+
+Before retweeting, make sure the tweet is not a retweet itself.
```

Comment:
For example, if you're using the Twitter Input Node with all the `#some_hashtag`, it will retweet the tweet with the `#some_hashtag`, and then, it will get it again, as a retweet.

If you will not filter it, it will try to retweet it again, and you will get an error "you already retweeted this".

Reply:
Can this check not be done within the node ?

Reply:
How do you suggest doing that?

We can accept the whole `tweet` object. If we want to keep it simple with just `retweet` with the tweet ID, I guess we need to use the Twitter API to first check if the tweet ID has already been retweeted by us. I think it is complicated, and I wanted to start with simpler solution.

Reply:
If we don’t know how to do it then advising the user that they should is just passing the buck isn't it ?

Reply:
I would suggest changing the wording from:
```
Before retweeting, make sure the tweet is not a retweet itself.
 ```
to
```
You cannot retweet a retweet. You can check if it is a retweet using the `msg.tweet.XYZ` property.
```
(assuming these is a property under `msg.tweet` that does the job)

Reply:
I will continue to work on it next week

Reply:
> If we don’t know how to do it then advising the user that they should is just passing the buck isn't it ?

_Originally posted by @dceejay in https://github.com/node-red/node-red-nodes/pull/835#discussion_r705933923_

Not exactly. I don't know and I didn't research exactly what is the right way to validate we are not retweeting the same tweet twice, so I leave it to the user to implement a custom workaround, based on its use case.

For example, in my use-case, I'm retweeting tweets by hashtag, and I filtering all the tweets from me, so that it will never try to retweet twice, since the second tweet will be always by me.

Reply:
> I would suggest changing the wording from:  
>```
>Before retweeting, make sure the tweet is not a retweet itself.
> ```
>to
>```
>You cannot retweet a retweet. You can check if it is a retweet using the `msg.tweet.XYZ` property.
>```
>(assuming these is a property under `msg.tweet` that does the job)

_Originally posted by @knolleary in https://github.com/node-red/node-red-nodes/pull/835#discussion_r705947899_

I'm very welcome to documentation improvements, my English is not so good. But about this specific suggestion, please see the previous comment. You **can** retweet a retweet, AFAIK, but you **can't** retweet tweet you already retweeted **yourself**. And I'm not sure how to check this (possibly by checking if one of the retweets of a tweets made by me...)